### ISSUE TITLE
Inconsistent error response with @auth decorated item

### ISSUE THREAD

**Issue by hello2all** [aws-amplify/amplify-cli#4290]
**Is your feature request related to a problem? Please describe.**

Given the following model with optional item `commentPublic` and `commentPrivate` decorated with `@auth` directive:

```
type Todo @model {
  id: ID!
  content: String!
  commentPublic: String
  commentPrivate: String
    @auth(rules: [ { allow: owner } ])
}
```

if we create an entry in db with optional items being `undefined`:

```
{
  "id": "id1",
  "comment": "test"
}
```

A query of the item above: `getTodo(id: "id1")` will return with an error regarding `@auth` decorated `commentPrivate`, but no error for `commentPublic` without `@auth` directive:

```
{
   "data":{
      "getTodo":{
         "id":"id1",
         "content":"test",
         "commentPrivate":null,
         "commentPublic":null
      }
   },
   "errors":[
      {
         "path": ["getTodo", "commentPrivate"],
         "message":"Unable to convert $util.toJson($context.source.commentPrivate) to class com.amazonaws.deepdish.transform.model.lambda.LambdaVersionedConfig."
      }
   ]
}
```

**Describe the solution you'd like**
Error response should be consistent regardless of `@auth` directive. I think this error should be suppressed when item is undefined.

**Describe alternatives you've considered**

An alternative is also throw error on `commentPublic` to ensure strict type definition, however I think that might be too wordy.

**Comment by attilah** [comment]
@hello2all Thanks for reporting this issue. The error message you seeing is not related to the root cause. I was able to reproduce this, please confirm the these were the steps you did:
- Create the schema with `@auth` rule on the type
- Execute `createTodo` mutation
- Move `@auth` to field
- Execute `getTodo` query

The reason I'm asking is that you ended up with invalid state because if you start with the schema you posted the CLI (and this can be a bug) does not create the `owner` field in the `Todo` type, so you have no way of populating it. But if you have the `owner` rule on the type the CLI adds the `owner` field to the type and works as expected. In general you cannot own a field in an item, so this `@auth` rule is not correct in this sense. I'll take it to the team for review to decide on this edge case.

**Comment by hello2all** [comment]
@attilah Thank you for the response. Sorry I made a mistake in the example that I gave.

I did use `owner` field but forget to included it  in the example. In the steps you gave I did the second step different, due to additional validation requirements I did not use the `createTodo` mutation, instead I used a custom lambda function resolver invoking a `PUT` operation to DynamoDB using with AWS SDK:

```javascript
const AWS = require('aws-sdk');
const ddb = new AWS.DynamoDB.DocumentClient({ apiVersion: '2012-08-10' });
  // some custom validations I did
  // validation...
 
  let putParam = {
    TableName: 'TodoTable',
    Item: {
      'id': id,
      'owner': ctx.identity.sub,
      'content': ctx.arguments.input
      // please note here I did not create the keys for commentPublic nor commentPrivate
    }
  }
  await ddb.put(putParam).promise();
```
Then execute `getTodo` query

Then I got the error msg:

```
   "data":{
      "getTodo":{
         "id":"id1",
         "owner": "ownerIdHere",
         "content":"test",
         "commentPrivate":null,
         "commentPublic":null
      }
   },
   "errors":[
      {
         "path": ["getTodo", "commentPrivate"],
         "message":"Unable to convert $util.toJson($context.source.commentPrivate) to class com.amazonaws.deepdish.transform.model.lambda.LambdaVersionedConfig."
      }
   ]
```

**Comment by SwaySway** [comment]
Hello @hello2all 
I see you are using the `ctx.identity.sub` as the value for owner. This might have an error as the owner field will default to looking into `cognito:username` within the identity token. You would need that to be in sync you can change it to looking into sub by changing the identityClaim Value (https://docs.amplify.aws/cli/graphql-transformer/directives#custom-claims) or changing the lambda to look for the username in the token.

As a takeaway on our side we will look into adding the `owner` field when specifying an owner rule.

**Comment by hello2all** [comment]
@SwaySway Thank you, I will look into it.

**Comment by SwaySway** [comment]
Closing this issue as the owner field does get added to the type when using `owner` based auth. Should you still experience any issues please comment below.
